### PR TITLE
support relatedTarget property for focus events

### LINKED ISSUE #2197: IE11 service worker null reference exception
Receiving 'Unable to get property 'register' of undefined or null reference' error.

image

This code in native-method.ts seems to be missing a null check:

        try {
            this.registerServiceWorker        = win.navigator.serviceWorker.register;
            this.getRegistrationServiceWorker = win.navigator.serviceWorker.getRegistration;
        }
        catch (e) {
            this.registerServiceWorker        = null;
            this.getRegistrationServiceWorker = null;
        }

### PR DESCRIPTION
**Pull request by AlexKamaev**
Based on https://github.com/DevExpress/testcafe/issues/2197

### CODE REVIEW

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/event/focus-blur.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ export default class FocusBlurSandbox extends SandboxBase {
             this._restoreElementNonScrollableParentsScrollState(this.scrollState.elementNonScrollableParentsScrollState);
     }
 
-    _raiseEvent (el, type, callback, withoutHandlers, isAsync, forMouseEvent, preventScrolling) {
+    _raiseEvent (el, type, callback, withoutHandlers, isAsync, forMouseEvent, preventScrolling, relEl) {
```
Comment (miherlosev) [inline src/client/sandbox/event/focus-blur.js]: Rename parameter: `relEl` -> `relatedTarget`

Comment (AlexKamaev) [reply src/client/sandbox/event/focus-blur.js]: in context of this issue or create separate one?

Comment (miherlosev) [reply src/client/sandbox/event/focus-blur.js]: in context

**Comment by miherlosev** [comment]
> There is a problem with our server tests in the latest node version - 9.10.0.
> @LavrovArtem is fixing it.

Done. Rebase with upstream

File: src/client/sandbox/event/focus-blur.js (hunk 1 above)
Comment (AlexKamaev) [reply src/client/sandbox/event/focus-blur.js]: Hammerhead already supports `relatedTarget` property for mouse and required drag events. It works for mouse events in testcafe, but does not work for drag events. So support of drag events should be fixed in testcafe repo

Comment (AlexKamaev) [reply src/client/sandbox/event/focus-blur.js]: I've created an issue https://github.com/DevExpress/testcafe/issues/2271 in testcafe repository

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
Rebase with upstream

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by AlexKamaev** [comment]
@testcafe-build-bot \retest

**Comment by miherlosev** [comment]
@testcafe-build-bot \retest

**Pull request merged**